### PR TITLE
Update to ImagePiiVerifyEngine class

### PR DESCRIPTION
**Pull request by niwilso**
## Change Description

Updating the ImagePiiVerifyEngine class to enable passing of a non-default analyzer engine into it. This is needed for cases where a custom recognizer is used.

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [ ] I have signed the CLA
- [ ] My code includes unit tests
- [ ] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

File: presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py [hunk 1]
```diff
@@ -18,22 +18,24 @@ def fig2img(fig):
 class ImagePiiVerifyEngine:
     """ImagePiiVerifyEngine class only supporting Pii verification currently."""
 
-    def __init__(self):
-        self.analyzer_engine = ImageAnalyzerEngine()
+    def __init__(self, analyzer_engine: ImageAnalyzerEngine = None):
```
Comment (omri374) [inline presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:21]: nit:
```suggestion
    def __init__(self, analyzer_engine: Optional[ImageAnalyzerEngine] = None):
```

**Review by omri374 — APPROVED** [review]

File: presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def fig2img(fig):
 class ImagePiiVerifyEngine:
     """ImagePiiVerifyEngine class only supporting Pii verification currently."""
 
-    def __init__(self, analyzer_engine: ImageAnalyzerEngine = None):
+    def __init__(self, analyzer_engine: Optional[ImageAnalyzerEngine] = None):
```
Comment (omri374) [inline presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:22]: @niwilso don't forget to add `from typing import Optional` :)

**Review by omri374 — APPROVED** [review]
Thanks! Great contribution

**Review by guybartal — APPROVED** [review]
LGTM

**Pull request merged**